### PR TITLE
Demos will be used as functional tests

### PR DESCRIPTION
**Pull request by lmilbaum**
As a first step, renaming demos folder to tests/functional. Eventually, those demos
will be used a functional tests in the CI.
Added a functional-tests quality gate to CI. For now it returns always passes. In a future PR, the demos will be invoked by the functional-tests make target.

### CODE REVIEW

**Comment by mlieberman85** [comment]
Will review on Monday a bit closer but I do like the general flow.

**Review by rgreinho — DISMISSED** [review]
👍

**Comment by bradbeck** [comment]
For some reason GitHub doesn't seem to like the new `ci.yaml` file.
https://github.com/buildsec/ssf/actions/runs/2180952201

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -78,3 +81,17 @@ jobs:
           cosign verify-attestation --key k8s://tekton-chains/signing-secrets "${IMAGE_URL}"
 
           kubectl wait --timeout=5m --for=condition=ready pods -l app=picalc -n prod
+
+ tests:
```
Comment (rgreinho) [inline .github/workflows/ci.yaml:85]: The ` tests:` job is not indented correctly, causing the CI to fail.

Comment (bradbeck) [inline .github/workflows/ci.yaml:85]:
```suggestion

  tests:
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -20,7 +20,10 @@ jobs:
         run: make lint
 
   setup:
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.platform }}
```
Comment (bradbeck) [inline .github/workflows/ci.yaml:26]: I'd suggest setting this aside for its own PR and discussion.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (lmilbaum) [reply .github/workflows/ci.yaml]: Thanks. Pushed the change.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (lmilbaum) [reply .github/workflows/ci.yaml:26]: Sure. I'll take care of it.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (lmilbaum) [reply .github/workflows/ci.yaml]: Fixed

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -78,3 +78,14 @@ jobs:
           cosign verify-attestation --key k8s://tekton-chains/signing-secrets "${IMAGE_URL}"
 
           kubectl wait --timeout=5m --for=condition=ready pods -l app=picalc -n prod
+
+tests:
```
Comment (bradbeck) [inline .github/workflows/ci.yaml:82]:
```suggestion
  tests:
```

**Review by bradbeck — APPROVED** [review]

**Comment by lmilbaum** [comment]
What is blocking us from review the PR?

**Comment by rgreinho** [comment]
> What is blocking us from review the PR?

It has been reviewed. The CI workflow is still broken and needs to be fixed.

**Comment by rgreinho** [comment]
Since the demo folder had been removed, we can close this PR. Feel free to re-open if it makes sense.

**Pull request closed without merging**